std/container/slist.d

```diff
@@ -589,6 +602,39 @@ Complexity: $(BIGOH n)
 
 /// ditto
     alias stableLinearRemove = linearRemove;
+
+/**
+Removes an element from the list in linear time.
+
+Complexity: $(BIGOH n)
+     */
+    void linearRemoveElement(T value)
+    {
+        enforce(_first);
```

Comment:
Why is removing `0` invalid?

Reply:
Also I think it would be better to return a boolean value - `true` iff the element existed and was successfully removed (`false` otherwise).